### DISCUSSION TITLE
Default heading size based on 'as' value

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [chakra-ui/chakra-ui#6325] [Q&A]
Is it possible to set default `size` based on another prop, e.g. `as` value?

I have the following scale:

```tsx
<Heading as="h1" size="lg">x</Heading>
<Heading as="h2" size="md">x</Heading>
<Heading as="h3" size="sm">x</Heading>
```

I'd like the next code to have the same effect:

```tsx
<Heading as="h1">x</Heading>
<Heading as="h2">x</Heading>
<Heading as="h3">x</Heading>
```

but still have an option to pass `size` to override the default scale:

```tsx
<Heading as="h1" size="xl">x</Heading>
<Heading as="h2" size="lg">x</Heading>
<Heading as="h3" size="md">x</Heading>
```

Is it possible to set up on the Theme level – without making new components?

**Comment by arzitzow** [comment]
I think this is a bug.

Here is a code sandbox: https://codesandbox.io/s/wonderful-sky-c6myqy?file=/src/index.tsx

The same `textStyles` applied to `Heading`, `Text` & `Box`.

Futhermore, should it really go back to the base if it's not defined? Shouldn't the breakpoints larger than `md` inherit the `md` defined styles? 🤔 

<img width="1229" alt="Screenshot 2023-08-29 at 6 22 01 PM" src="https://github.com/chakra-ui/chakra-ui/assets/126702954/dfdc9443-6937-4faf-a8ff-c58888ddc3f2">
<img width="826" alt="Screenshot 2023-08-29 at 6 21 52 PM" src="https://github.com/chakra-ui/chakra-ui/assets/126702954/94c3279f-5037-4d58-b2eb-3387b0dfbbf8">
<img width="493" alt="Screenshot 2023-08-29 at 6 21 45 PM" src="https://github.com/chakra-ui/chakra-ui/assets/126702954/b6859497-2085-45a7-8578-53bc47d08566">
